dali/kernels/imgproc/resample/resampling_impl_cpu.h

```diff
@@ -0,0 +1,272 @@
+  const float bias = std::is_integral<Out>::value ? 0.5f : 0;
+  const int channels = static_channels < 0 ? dynamic_channels : static_channels;
+
+  int x0 = in_columns[x];
+  int k0 = x * support;
+
+  if (static_channels < 0) {
+    // we don't know how many channels we have at compile time - inner loop over filter
+    for (int c = 0; c < channels; c++) {
+      float sum = bias;
+      for (int k = 0; k < support; k++) {
+        int srcx = x0 + k;
+        if (clamp_left) if (srcx < 0) srcx = 0;
+        if (clamp_right) if (srcx > w-1) srcx = w-1;
+        sum += coeffs[k0 + k] * in[srcx * channels + c];
+      }
+      out[channels * x + c] = clamp<Out>(sum);
+    }
+  } else {
+    // we know how many channels we have at compile time - inner loop over channels
+    float tmp[static_channels > 0 ? static_channels : 1];  // NOLINT
+    for (int c = 0; c < channels; c++)
+      tmp[c] = bias;
+
+    for (int k = 0; k < support; k++) {
+      int srcx = x0 + k;
+      if (clamp_left) if (srcx < 0) srcx = 0;
+      if (clamp_right) if (srcx > w-1) srcx = w-1;
+      for (int c = 0; c < channels; c++) {
+        tmp[c] += coeffs[k0 + k] * in[srcx * channels + c];
+      }
+    }
+
+    for (int c = 0; c < channels; c++)
+      out[channels * x + c] = clamp<Out>(tmp[c]);
+  }
+}
+
+template <int static_channels = -1, typename Out, typename In>
+void ResampleHorz_Channels(
+    Surface2D<Out> out, Surface2D<In> in, const int *in_columns,
+    const float *coeffs, int support) {
+  const int channels = static_channels < 0 ? out.channels : static_channels;
+
+  int first_regular_col = 0;
+  int last_regular_col = out.width - 1;
+  while (first_regular_col < out.width && in_columns[first_regular_col] < 0)
+    first_regular_col++;
+  while (last_regular_col >= 0 && in_columns[last_regular_col] + support > in.width)
+    last_regular_col--;
+
+  for (int y = 0; y < out.height; y++) {
+    Out *out_row = &out(0, y);
+    const In *in_row = &in(0, y);
+
+    int x = 0;
+
+    for (; x < first_regular_col && x <= last_regular_col; x++) {
+      ResampleCol<static_channels, true, false>(
+        out_row, in_row, x, in.width, in_columns, coeffs, support, channels);
+    }
+    for (; x < first_regular_col; x++) {
+      ResampleCol<static_channels, true, true>(
+        out_row, in_row, x, in.width, in_columns, coeffs, support, channels);
+    }
+    for (; x <= last_regular_col; x++) {
+      ResampleCol<static_channels, false, false>(
+        out_row, in_row, x, in.width, in_columns, coeffs, support, channels);
+    }
+    for (; x < out.width; x++) {
+      ResampleCol<static_channels, false, true>(
+        out_row, in_row, x, in.width, in_columns, coeffs, support, channels);
+    }
+  }
+}
+
+template <typename Out, typename In>
+void ResampleVert(
+    Surface2D<Out> out, Surface2D<In> in, const int32_t *in_rows,
+    const float *row_coeffs, int support) {
+  const float bias = std::is_integral<Out>::value ? 0.5f : 0;
+  const int tile = 64;
```

Comment:
```suggestion
  constexpr int tile = 64;
```

Reply:
It makes cpplint cry that tmp[tile] is a "variable-sized array" and advises renaming it to `kTile` which is ugly :(

Reply:
`// NOLINT` :smiley:  ?